File: from-3201-to-3300/3216-lexicographically-smallest-string-after-a-swap/solution.go
func getSmallestString(s string) string {
    res := []byte(s)
    for i := 1; i < len(s); i++ {
        a, b := s[i - 1], s[i]
        if a % 2 == b % 2 && a > b {
            res[i - 1], res[i] = res[i], res[i - 1]
            return string(res)
        }
    }
    return s
}

File: from-3201-to-3300/3216-lexicographically-smallest-string-after-a-swap/solution.py
class Solution:
    def getSmallestString(self, s: str) -> str:
        lst = [int(a) for a in list(s)]
        for i in range(len(s) - 1):
            if lst[i] % 2 == lst[i + 1] % 2 and lst[i] > lst[i + 1]:
                lst[i], lst[i + 1] = lst[i + 1], lst[i]
                return "".join([str(a) for a in lst])
        return s